fastlane/lib/fastlane/plugins/plugin_update_manager.rb

```diff
@@ -0,0 +1,65 @@
+module Fastlane
+  # Alert the user when updates for plugins are available
+  class PluginUpdateManager
+    def self.start_looking_for_updates
+      Thread.new do
+        self.references.each do |plugin_name, current_plugin|
+          begin
+            self.server_results[plugin_name] = Gem::Version.new(fetch_latest_version(plugin_name))
+          rescue
+          end
+        end
+      end
+    end
+
+    def self.show_update_status
+      # We set self.server_results to be nil
+      # this way the table is not printed twice
+      # (next to the summary table or when an exception happens)
+      return unless self.server_results.count > 0
+
+      rows = []
+      self.references.each do |plugin_name, current_plugin|
+        latest_version = self.server_results[plugin_name]
+        next if latest_version.nil?
+        current_version = Gem::Version.new(current_plugin[:version_number])
+        next if current_version >= latest_version
+
+        rows << [
+          plugin_name.gsub(PluginManager.plugin_prefix, ''),
+          current_version.to_s.red,
+          latest_version.to_s.green
+        ]
+      end
+
+      if rows.empty?
+        UI.verbose("All plugins are up to date")
+        return
+      end
+
+      puts Terminal::Table.new({
+        rows: rows,
+        title: "Plugin updates available".yellow,
+        headings: ["Plugin", "Your Version", "Latest Version"]
+      })
+      UI.message "To update all plugins, just run"
+      UI.command "fastlane update_plugins"
+      puts ''
+      @server_results = nil
+    end
+
+    def self.references
+      Fastlane.plugin_manager.plugin_references
+    end
+
+    def self.fetch_latest_version(gem_name)
+      PluginManager.fetch_gem_info_from_rubygems(gem_name)["version"]
```

Comment:
Should this just return a `Gem::Version` instead of a string? It looks like line 8 is the only place it is used (in this file at least) and it could be nicer to make line 8 a little cleaner?